### PR TITLE
Add git repo validation and build tool detection using git-service

### PR DESCRIPTION
**Pull request by rohitkrai03**
Related Story - https://jira.coreos.com/browse/ODC-1155

This PR - 
- Adds validation of git repo using git service.
- Adds detection and recommendation of builder images based on the git repo using git service.

Gif - 
ezgif com-video-to-gif

### CODE REVIEW

**Comment by rohitkrai03** [comment]
> For the first-time builder image recommends by applying the border and star icon on the corner of the card. But when I change the git URL then star icon move to another detected builder image but border remains to the previous detected builder image.
> 
> Peek 2019-10-15 12-15

This is desired behaviour since we do not want to select a builder image if user has already selected one. In this case it was pre-selected by recommendation earlier.

**Comment by vikram-raj** [comment]
> This is desired behaviour since we do not want to select a builder image if user has already selected one. In this case it was pre-selected by recommendation earlier.

@rohitkrai03 I am selecting the builder image. For the first time border get applied automatically.
Is the border applied only when the user selects the builder image?

**Comment by rohitkrai03** [comment]
lo

> > This is desired behaviour since we do not want to select a builder image if user has already selected one. In this case it was pre-selected by recommendation earlier.
> 
> @rohitkrai03 I am selecting the builder image. For the first time border get applied automatically.
> Is the border applied only when the user selects the builder image?

If user has not selected any builder image then while recommending we also select the recommended builder image, this is why you see blue border at first since you have not selected anything. But the second time you already have a selected builder image that was selected by recommendation, since we cannot distinguish if it was a user selection or recommendation selection for the second time, the builder image is not selected again. It retains the old value.

**Comment by rohitkrai03** [comment]
> Getting this error
> `Uncaught (in promise) TypeError: Cannot read property 'isRepoReachable' of null`
> 
> When I leave the Git URL field without entering any value.

Fixed

**Comment by vikram-raj** [comment]
@rohitkrai03 No, as per this PR we are not selecting the builder image until the user selects one. Because I am not seeing the builder image version dropdown.

**Comment by rohitkrai03** [comment]
> @rohitkrai03 No, as per this PR we are not selecting the builder image until the user selects one. Because I am not seeing the builder image version dropdown.

Fixed. I missed the tag selection for builder image earlier.

**Comment by debsmita1** [comment]
At first I entered `https://github.com/rails/rails` as the git url and on validation it detects `Ruby` builder image. On updating the url I saw this :
validating

I feel we shouldn't show the `Builder Image detected` notification when the url is being validated. What do you think ?

**Comment by debsmita1** [comment]
Do you think we should hide the notification if user selects a builder image after being notified ?

user selection

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 1]
```diff
@@ -31,13 +50,44 @@ const GitSection: React.FC<GitSectionProps> = ({ showSample }) => {
...
+        }
+      } else {
+        setFieldValue('image.couldNotRecommend', true);
+        setFieldValue('image.recommended', '');
+      }
+    } else {
+      setFieldValue('image.recommended', '');
+      setFieldValue('image.couldNotRecommend', false);
+      setFieldValue('git.isUrlValidated', false);
+      setFieldError('git.url', 'Git respository is not reachable');
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx]:
```suggestion
      setFieldError('git.url', 'Git respository is not reachable.');

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/git/GitSection.tsx]: Done

**Comment by rohitkrai03** [comment]
@debsmita1 Both of your comments related to alert notifications should be decided by UX and since I couldn't find anything related to this in the design doc i think its okay for now. If UX wants us to change this behaviour we can add it later as a fix.

**Comment by christianvogt** [comment]
I think we should start clearing out the validation and recommended states when the user goes back and edits the git url. On change before they even blur the field.

**Comment by christianvogt** [comment]
Validation message doesn't match design:
image
image

Recommended star color is very dark compared to designs:
image
image

**Comment by rohitkrai03** [comment]
> I think we should start clearing out the validation and recommended states when the user goes back and edits the git url. On change before they even blur the field.

Done.

**Comment by rohitkrai03** [comment]
> Validation message doesn't match design:
> image
> image
> 
> Recommended star color is very dark compared to designs:
> image
> image

Updated the color of star icon from `--pf-global--primary-color--200` to `--pf-global--primary-color--100`. For the validation texts below git url input field I've added the check icon temporarily until https://github.com/patternfly/patternfly-next/issues/2320 is implemented. I've made the text bolder for now.

**Pull request merged**